### ISSUE TITLE
possible missing CRYPTSP.dll in GD3D11-17.7-dev15-fix1

### ISSUE THREAD

**Issue by stefson** [Kirides/GD3D11#28]
So, after some further testing: d3dcompiler_47.dll is actually a hard dependency and needed at runtime. The newly provided d3dcompiler_47.dll with x86 flavour now loads fine in wine, but there's a minor show stopper: 

```
0124:err:module:import_dll Library CRYPTSP.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\d3dcompiler_47.dll") not found
0124:err:module:import_dll Library d3dcompiler_47.dll (which is needed by L"C:\\windows\\system32\\d3d10.dll") not found
0124:err:module:find_forwarded_export module not found for forward 'd3d10.D3D10CreateStateBlock' used by L"C:\\windows\\system32\\d3d10_1.dll"
0124:err:module:import_dll Library CRYPTSP.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\d3dcompiler_47.dll") not found
0124:err:module:import_dll Library d3dcompiler_47.dll (which is needed by L"C:\\windows\\system32\\d3d10.dll") not found
0124:err:module:find_forwarded_export module not found for forward 'd3d10.D3D10StateBlockMaskEnableAll' used by L"C:\\windows\\system32\\d3d10_1.dll"
0124:err:module:import_dll Library CRYPTSP.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\D3DCOMPILER_47.dll") not found
0124:err:module:import_dll Library D3DCOMPILER_47.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\DDRAW.dll") not found
0124:err:module:import_dll Library DDRAW.dll (which is needed by L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\Gothic2.exe") not found
0124:err:module:LdrInitializeThunk Importing dlls for L"C:\\Program Files\\JoWood\\Gothic II Gold\\System\\Gothic2.exe" failed, status c0000135
```

what do you think?

**Comment by Kirides** [comment]
Is there a way to install it using winetricks?

**Comment by stefson** [comment]
I've double checked current master winetricks, and I believe it is not possible to use winetricks for that.

**Comment by Kirides** [comment]
What happens if you just remove the d3dcompiler_47.dll from the Gothic\System directory and use winetricks to install it into the wine snapshot?

**Comment by stefson** [comment]
I think the cryptsp.dll is an additional dependency, which came from slightly updating d3dcompiler_47.dll between 17.7-dev11-fix2 and 17.8-dev15-fix1 - I'm on the linux machine now, but I saw slight version difference in the file properties of the dll when inspecting them on win7

It's absolutly possible to remove or rename d3dcompiler_47.dll from the System directory, and install it instead via winetricks. This seems to have been the standard procedere all the time, as people advised me to try it that way to solve the problem with the x86_64 missmatched dll

It's also possible to pick the d3dcompiler_47.dll from from 17.7-dev11-fix2 update instead, to avoid the additional cryptsp.dll

I don't know, pick your poison? :-)

**Comment by Kirides** [comment]
Well, using an unsupported way to run the game lead to needing unconventional ways to run it sometimes

**Comment by stefson** [comment]
the problem with winetricks is, it needs a downloadable and extractable file to get the dll from. the script can't unpack .msu files, so it picks an old version of firefox-62 to extract d3dcompiler_47.dll from.

**Comment by stefson** [comment]
this is most likely fixed in 17.7-dev16, as you downgraded the d3dcompiler_47.dll in https://github.com/Kirides/GD3D11/commit/87dac7d218f64e0df6680499b8311d0446f63e03

if you push out the binaries I'll gladly confirm.

**Comment by stefson** [comment]
fixed in v17.7-dev16
